pyiceberg/table/__init__.py

```diff
@@ -3253,6 +3253,15 @@ def __init__(self, tbl: Table) -> None:
         except ModuleNotFoundError as e:
             raise ModuleNotFoundError("For metadata operations PyArrow needs to be installed") from e
 
+    def _snapshot(self, snapshot_id: Optional[int] = None) -> Optional[Snapshot]:
+        if snapshot_id:
```

Comment:
Nit. Otherwise, it will fail on Snapshot ID 0.
```suggestion
        if snapshot_id is not None:
```